azure-devops/provision-agent.ps1

```diff
@@ -0,0 +1,204 @@
+        $args = @('/i', $msiPath, '/norestart', '/quiet', '/qn')
+
+        curl.exe -L -o $msiPath $Uri
+        Write-Output "Installing $Name..."
+        $proc = Start-Process -FilePath $msiExecPath -ArgumentList $args -Wait -PassThru
+        $exitCode = $proc.ExitCode
+        if ($exitCode -eq 0 -or $exitCode -eq 3010)
+        {
+            Write-Output 'Installation successful!'
+        }
+        else
+        {
+            Write-Output "Nonzero exit code returned by the installation process : $exitCode."
+            exit $exitCode
+        }
+    }
+    catch
+    {
+        Write-Output "Failed to install $Name!"
+        Write-Output $_.Exception.Message
+        exit -1
+    }
+}
+
+Function InstallZip
+{
+    Param(
+        [String]$Name,
+        [String]$Uri,
+        [String]$Dir
+    )
+
+    try
+    {
+        Write-Output "Downloading $Name..."
+        [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+        [string]$zipPath = $randomRoot + '.zip'
+        curl.exe -L -o $zipPath $Uri
+        Write-Output "Installing $Name..."
+        Expand-Archive -Path $zipPath -DestinationPath $Dir -Force
+    }
+    catch
+    {
+        Write-Output "Failed to install $Name!"
+        Write-Output $_.Exception.Message
+        exit -1
+    }
+}
+
+Function InstallLLVM
+{
+    Param(
+        [String]$Uri
+    )
+
+    Write-Output 'Downloading LLVM...'
+    [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+    [string]$installerPath = $randomRoot + '.exe'
+
+    curl.exe -L -o $installerPath $Uri
+    Write-Output 'Installing LLVM...'
+    $proc = Start-Process -FilePath $installerPath -ArgumentList @('/S') -NoNewWindow -Wait -PassThru
+    $exitCode = $proc.ExitCode
+
+    if ($exitCode -eq 0 -or $exitCode -eq 3010)
+    {
+        Write-Output 'Installation successful!'
+    }
+    else
+    {
+        Write-Output "Nonzero exit code returned by the installation process : $exitCode."
+        exit $exitCode
+    }
+}
+
+Function InstallPython
+{
+    Param(
+        [String]$Uri
+    )
+
+    Write-Output 'Downloading Python...'
+    [string]$randomRoot = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+    [string]$installerPath = $randomRoot + '.exe'
+
+    curl.exe -L -o $installerPath $Uri
+    Write-Output 'Installing Python...'
+    $proc = Start-Process -FilePath $installerPath -ArgumentList @('/passive', 'InstallAllUsers=1', 'PrependPath=1', 'CompileAll=1') -Wait -PassThru
+    $exitCode = $proc.ExitCode
+
+    if ($exitCode -eq 0)
+    {
+        Write-Output 'Installation successful!'
+    }
+    else
+    {
+        Write-Output "Nonzero exit code returned by the installation process : $exitCode."
+        exit $exitCode
+    }
+}
+
+if ([string]::IsNullOrEmpty($PersonalAccessToken) -or ($PersonalAccessToken -eq 'CHANGE THIS')) {

```

Comment:
In `install-scale-set-extension.ps1` it was defaulted to `PERSONAL_ACCESS_TOKEN`, not `CHANGE THIS`. (Nor could spaces survive, because `provision-agent-bootstrap.cmd` didn't use `"%2"`.)